fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/NumericAggregationValue.java

```diff
@@ -591,7 +674,33 @@ public enum PhysicalOperator {
         MAX_I(LogicalOperator.MAX, TypeCode.INT, TypeCode.INT, Objects::requireNonNull, (s, v) -> Math.max((int)s, (int)v), identity()),
         MAX_L(LogicalOperator.MAX, TypeCode.LONG, TypeCode.LONG, Objects::requireNonNull, (s, v) -> Math.max((long)s, (long)v), identity()),
         MAX_F(LogicalOperator.MAX, TypeCode.FLOAT, TypeCode.FLOAT, Objects::requireNonNull, (s, v) -> Math.max((float)s, (float)v), identity()),
-        MAX_D(LogicalOperator.MAX, TypeCode.DOUBLE, TypeCode.DOUBLE, Objects::requireNonNull, (s, v) -> Math.max((double)s, (double)v), identity());
+        MAX_D(LogicalOperator.MAX, TypeCode.DOUBLE, TypeCode.DOUBLE, Objects::requireNonNull, (s, v) -> Math.max((double)s, (double)v), identity()),
+        BITMAP_LL(LogicalOperator.BITMAP, TypeCode.LONG, TypeCode.BYTES,
+                s -> {
+                    BitSet sset = new BitSet();
```

Comment:
I think we have a discrepancy between this and what the index maintainer returns in case of using an indexed version of the bitmap.

If the bit bucket size is for example 1000, and we have only the first bit set:
- the indexed bitmap version will return 1000 bits with the first bit set.
- the streaming aggregate version will return only 1 bit.

I do not think this is ok. OTOH, I am not exactly sure how to fix it, is it e.g. possible to pre-allocate the bitset with a specific size?

Reply:
My proposal is, for the result byte array:
```
if res.length > BitmapValueIndexMaintainer.MAX_ENTRY_SIZE {
    throw ex;
} else if (res.length > BitmapValueIndexMaintainer.DEFAULT_ENTRY_SIZE {
    return res;
} else {
    return Arrays.copyOf(res, DEFAULT_ENTRY_SIZE);
}
```
wdyt?

Reply:
I don't think there is necessarily a good solution here, unless we control the DDL/DQL completely, because it is possible to create an aggregate bitmap index using a custom bucket size with:

https://github.com/FoundationDB/fdb-record-layer/blob/fb3c634e4c1bb8101873f1e9ecb2d8fa7aa16538/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/IndexOptions.java#L157

therefore, it is impossible to align these, so your solution is what I would do as well.